qiskit_ibm_runtime/session.py

```diff
@@ -112,8 +113,15 @@ def __init__(
         else:
             self._service = service
 
-        if self._service.channel == "ibm_quantum" and not backend:
-            raise ValueError('"backend" is required for ``ibm_quantum`` channel.')
+        if not backend:
+            if self._service.channel == "ibm_quantum":
+                raise ValueError('"backend" is required for ``ibm_quantum`` channel.')
+            warnings.warn(
+                "In a future release no sooner than 3 months after the release date of "
+                "qiskit-ibm-runtime version 0.21.0, "
+                "``backend`` will no longer be optional when using primitives. "
+                "Please provide a backend."
+            )
```

Comment:
Why not use the `issue_deprecation_msg()` so the format is consistent and sets the right type.

Reply:
I didn't think the wording in `issue_deprecation_msg()` fit well here 

```f"and will be removed no sooner than {period} after the release date. {remedy}"``` 

I can update the message to more closely match that format and also set it as `DeprecationWarning`